docs/guides/models.md

```diff
@@ -97,8 +98,12 @@ const machine = createMachine(
   },
   {
     actions: {
-      updateAge: userModel.assign({
-        age: (_, event) => event.value // inferred
+      updateAge: userModel.assign((_, event) => {
+        // the event cannot be inferred automatically here, so we check it manually
```

Comment:
I know we explain later how to use the 2nd arg to `model.assign` to type the event, but that doesn't come for another 100 lines. Feels like we might be introducing a more inconvenient way to handle this first.

Reply:
The second argument doesn't work for named actions, though >_>

You would have to write:
```typescript
actions: {
  updateAge: userModel.assign({
    age: (_, event) => event.age
  }, 'updateAge') as any
}
```
which I believe is markedly worse.

Reply:
Are you sure? I use it all the time!

Reply:
Ohhhhh #2122 probably applies here. I've started setting `strictFunctionTypes: false` on all my XState projects to avoid errors in this situation for the time being, but we probably shouldn't recommend that.

Reply:
It's a similar but different problem to #2122

When you use the second argument of `model.assign()`, you get a function that takes only the event with the narrowed type.  This works if used in `on: { someEvent: { actions: model.assign(..., 'someEvent') } }` - because TS expect this action definition to be for this specific event.

However, in general actions configuration, TS expects the action to be able to handle ALL possible machine events because it cannot infer which actual events call this action.

Reply:
Ahhhh okay, I see what you mean. `strictFunctionTypes: false` disables the check that cause either error. https://codesandbox.io/s/exciting-diffie-xbmf8?file=/src/App.tsx